codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/error/ServiceErrorGenerator.kt

```diff
@@ -80,7 +80,16 @@ class ServiceErrorGenerator(
                         errors.map { it.id },
                     )
                 }
-            rust("impl #T for Error {}", RuntimeType.StdError)
+            rustBlock("impl #T for Error", RuntimeType.StdError) {
+                rustBlock("fn source(&self) -> std::option::Option<&(dyn #T + 'static)>", RuntimeType.StdError) {
+                    rustBlock("match self") {
+                        allErrors.forEach {
+                            rust("Error::${symbolProvider.toSymbol(it).name}(inner) => inner.source(),")
```

Comment:
hmm...should this be `inner.source()` or `Some(&inner as _)` — I think maybe the latter? `inner.source()` might be an extra level

Reply:
```
Unhandled error
: some error
```

looks more correct to me